blockcontact.php

```diff
@@ -24,145 +24,140 @@
-		$helper->table =  $this->table;
-		$lang = new Language((int)Configuration::get('PS_LANG_DEFAULT'));
-		$helper->default_form_language = $lang->id;
-		$helper->allow_employee_form_lang = Configuration::get('PS_BO_ALLOW_EMPLOYEE_FORM_LANG') ? Configuration::get('PS_BO_ALLOW_EMPLOYEE_FORM_LANG') : 0;
-		$this->fields_form = array();
-
-		$helper->identifier = $this->identifier;
-		$helper->submit_action = 'submitModule';
-		$helper->currentIndex = $this->context->link->getAdminLink('AdminModules', false).'&configure='.$this->name.'&tab_module='.$this->tab.'&module_name='.$this->name;
-		$helper->token = Tools::getAdminTokenLite('AdminModules');
-		$helper->tpl_vars = array(
-			'fields_value' => $this->getConfigFieldsValues(),
-			'languages' => $this->context->controller->getLanguages(),
-			'id_language' => $this->context->language->id
-		);
-
-		return $helper->generateForm(array($fields_form));
-	}
-
-	public function getConfigFieldsValues()
-	{
-		return array(
-			'blockcontact_telnumber' => Tools::getValue('blockcontact_telnumber', Configuration::get('BLOCKCONTACT_TELNUMBER')),
-			'blockcontact_email' => Tools::getValue('blockcontact_email', Configuration::get('BLOCKCONTACT_EMAIL')),
-		);
-	}
+    public function __construct()
+    {
+        $this->name = 'blockcontact';
+        $this->author = 'PrestaShop';
+        $this->tab = 'front_office_features';
+        $this->version = '2.0';
+
+        $this->bootstrap = true;
+        parent::__construct();
+
+        $this->displayName = $this->l('Contact block');
+        $this->description = $this->l('Allows you to add additional information about your store\'s customer service.');
+        $this->ps_versions_compliancy = array('min' => '1.7', 'max' => _PS_VERSION_);
+    }
+
+    public function install()
+    {
+        return parent::install()
+            && Configuration::updateValue('BLOCKCONTACT_TELNUMBER', '')
+            && Configuration::updateValue('BLOCKCONTACT_EMAIL', '')
+            && $this->registerHook('displayNav');
+    }
+
+    public function uninstall()
+    {
+        // Delete configuration
+        return Configuration::deleteByName('BLOCKCONTACT_TELNUMBER') && Configuration::deleteByName('BLOCKCONTACT_EMAIL') && parent::uninstall();
+    }
+
+    public function getContent()
+    {
+        $html = '';
+        // If we try to update the settings
+        if (Tools::isSubmit('submitModule'))
+        {
+            Configuration::updateValue('BLOCKCONTACT_TELNUMBER', Tools::getValue('blockcontact_telnumber'));
+            Configuration::updateValue('BLOCKCONTACT_EMAIL', Tools::getValue('blockcontact_email'));
+            $this->_clearCache('blockcontact.tpl');
+            $this->_clearCache('nav.tpl');
+            $html .= $this->displayConfirmation($this->l('Configuration updated'));
+        }
+
+        $html .= $this->renderForm();
+
+        return $html;
+    }
+
+    public function renderWidget($hookName = null, array $configuration = [])

```

Comment:
Not a big deal, but I think that under Strict Standards implementing the renderWidget method with optional arguments when the interface has them mandatory produces notices.